### PR TITLE
Moved and added helper functions

### PR DESCRIPTION
**Pull request by Bronek**
Moved some helper functions from `harness.go` to a new file, added new helper functions to this file.

### CODE REVIEW

File: pkg/test_framework/helpers.go [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+}
+
+type WaitSource func()(runtime.Object, error)
+
+func WaitForNone(timeout time.Duration, from WaitSource) error {
+	none := int32(0)
+	return WaitFor(&none, timeout, from)
+}
+
+func WaitFor(reps* int32, timeout time.Duration, from WaitSource) error {
```
Comment (mhaseebmlk) [inline pkg/test_framework/helpers.go]: Would it be useful to also have a [`WaitForResource`](https://sourcegraph.yelpcorp.com/services/kafka-operator/-/blob/acceptance/helpers.go#L82:6) helper function?

**Comment by Baisang** [comment]
> The logic seems okay... did you test these manually or are you planning to write unit tests later?

Oh, I see you've tested them internally with our flink operator :)

File: pkg/test_framework/helpers.go (hunk 1 above)
Comment (Bronek) [reply pkg/test_framework/helpers.go]: Honestly, I am not sure. Considering how you use that function in kafka operator acceptance tests, it is not clear to me at this moment what useful generalisation could look like. But **maybe** I will figure out something later.

**Comment by Bronek** [comment]
> > The logic seems okay... did you test these manually or are you planning to write unit tests later?
> 
> Oh, I see you've tested them internally with our flink operator :)

... also added unit tests now - thank you for keeping in on "straight and narrow" ;)

**Pull request merged**